java/src/main/java/ai/rapids/cudf/RegexProgram.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.rapids.cudf;
+
+/**
+ * Regex program class, closely following cudf::strings::regex_program.
+ */
+public class RegexProgram {
+  private String pattern;        // regex pattern
+  private RegexFlags flags;      // regex flags for interpreting special characters in the pattern
+  private CaptureGroups capture; // controls how capture groups in the pattern are used
+
+  /**
+   * Constructor for RegexProgram
```

Comment:
nit: Can we explicitly call out that the default is to extract a capture group? The reason I say this is because we want to match CUDF's defaults, but for many APIs this is not the ideal choice, and I want us to be careful with doing this.  As an alternative we could just provide a way for us to reset the capture group after it is created.

Reply:
Thank you very much for your review and comments! I have updated all the changes.